app/src/main/report/components/OneTimeLinkContext.tsx

```diff
@@ -1,73 +1,60 @@
 import * as React from "react";
-import {doNothing} from "../../shared/Helpers";
-import {connectToStores} from "../../shared/alt";
-import fetcher from "../../shared/sources/Fetcher";
 import {OneTimeToken} from "../models/OneTimeToken";
-import {oneTimeTokenStore} from "../stores/OneTimeTokenStore";
-import {oneTimeTokenActions} from "../actions/OneTimeTokenActions";
+import {connect, Dispatch} from "react-redux";
+import {ReportAppState} from "../reducers/reportAppReducers";
+import {oneTimeTokenActionCreators} from "../actionCreators/oneTimeTokenActionCreators";
+import {buildReportingURL} from "../services/AbstractReportLocalService";
 
 interface PublicProps {
     href: string;
     className?: string;
 }
 
-interface Props extends PublicProps {
-    token: OneTimeToken;
```

Comment:
So one thing to note is that I'm not using this `OnetimeToken` type anywhere, but just treating the token as a string to be retrieved and passed back to the API, without decoding/looking inside. I couldn't see a good reason to be doing the latter, but feel free to point out if I'm wrong about this?